lychee-lib/src/types/input.rs

```diff
@@ -414,6 +487,20 @@ impl Input {
     }
 }
 
+/// Helper function to check if a file path matches any of the given extensions
+fn file_extensions_match(path: &Path, extensions: &FileExtensions) -> bool {
+    // If the path has no extension, check if we accept plaintext files
+    // Note: We treat files without extensions as plaintext
+    if path.extension().is_none() {
+        return extensions.contains("txt") || extensions.contains("");
+    }
```

Comment:
This is a bit of a workaround for files without extensions. It works and probably matches the user intent, so I'd argue it's fine.